### PR TITLE
Fix the description of the postprocessing service

### PR DESCRIPTION
**Pull request by mmattel**
## Description
This PR fixes the descriptions of the new postprocessing service. It aligns the text where possible to existing or similar descriptions.

* Though I have improved the `POSTPROCESSING_VIRUSSCAN` desc - this still needs improvement as the text is not final. The reason / usecase setting this envvar is not clear.
* Same for `POSTPROCESSING_DELAY`. Why is this setting needed, do we also have ms (milliseconds) and what is the impact when set.

@kobergj can you add the missing pieces?

## Related Issue
none

## Motivation and Context
Make the envvars well described.

## How Has This Been Tested?
Text change only

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: services/postprocessing/pkg/config/config.go [hunk 1]
```diff
@@ -23,15 +23,15 @@ type Config struct {
 // Postprocessing definces the config options for the postprocessing service.
 type Postprocessing struct {
 	Events          Events        `yaml:"events"`
-	Virusscan       bool          `yaml:"virusscan" env:"POSTPROCESSING_VIRUSSCAN" desc:"should the system do a virusscan? Needs antivirus service"`
-	Delayprocessing time.Duration `yaml:"delayprocessing" env:"POSTPROCESSING_DELAY" desc:"the sytem sleeps for this time while postprocessing"`
+	Virusscan       bool          `yaml:"virusscan" env:"POSTPROCESSING_VIRUSSCAN" desc:"Needs as prerequisite the antivirus service to be enabled and configured. should the system do a virusscan?"`
+	Delayprocessing time.Duration `yaml:"delayprocessing" env:"POSTPROCESSING_DELAY" desc:"A delay in the given duration during the sytem will sleep while postprocessing. The duration can be set as number followed by a unit identifyer like s, m or h."`
```
Comment (phil-davis) [inline services/postprocessing/pkg/config/config.go:27]:
```suggestion
	Delayprocessing time.Duration `yaml:"delayprocessing" env:"POSTPROCESSING_DELAY" desc:"A delay in the given duration during the system will sleep while postprocessing. The duration can be set as number followed by a unit identifier like s, m or h."`
```

Comment (phil-davis) [reply services/postprocessing/pkg/config/config.go:27]: "A delay in the given duration during the system will sleep while postprocessing" - I don't understand this. If I know what it actually does, then I could suggest how to edit the words ;)

Comment (mmattel) [reply services/postprocessing/pkg/config/config.go:27]: You are completely right 👍 
I added this change to easily start reviewing and to add clarity as I have written in my PR description.

Comment (mmattel) [reply services/postprocessing/pkg/config/config.go:27]: You are completely right 👍 
I added this change to easily start reviewing and to add clarity as I have written in my PR description.

Comment (kobergj) [reply services/postprocessing/pkg/config/config.go:27]: Yeah I wasn't sure how to phrase this best. I'll try to describe:

During postprocessing the file will be in `425` state (processing) during which it cannot be downloaded. To be able to test this one can set to e.g. `30s`. Ocis will then sleep during postprocessing for 30 seconds. That means the minimum amount of time this file is in processing is 30s.

To be clear: This envvar enables developers to test their new features against file that are in processing. It is not meant to be set on a productive enviroment. (Unless as admin you want to annoy your users.

File: services/postprocessing/pkg/config/config.go [hunk 2]
```diff
@@ -23,15 +23,15 @@ type Config struct {
 // Postprocessing definces the config options for the postprocessing service.
 type Postprocessing struct {
 	Events          Events        `yaml:"events"`
-	Virusscan       bool          `yaml:"virusscan" env:"POSTPROCESSING_VIRUSSCAN" desc:"should the system do a virusscan? Needs antivirus service"`
-	Delayprocessing time.Duration `yaml:"delayprocessing" env:"POSTPROCESSING_DELAY" desc:"the sytem sleeps for this time while postprocessing"`
+	Virusscan       bool          `yaml:"virusscan" env:"POSTPROCESSING_VIRUSSCAN" desc:"Needs as prerequisite the antivirus service to be enabled and configured. should the system do a virusscan?"`
```
Comment (kobergj) [inline services/postprocessing/pkg/config/config.go:26]: This is actually correct. Additionally to the antivirus service being running this envvar needs to be set to `true` to enable virusscanning during postprocessing.

An alternative would be to have this set to `false` and trigger virus scans only manually or on download.

Comment (mmattel) [reply services/postprocessing/pkg/config/config.go:26]: In a nutshell, we are doing something afterwards (postprocessin) while we are doing it right now (upload). Absolutely not confusing and totally clear 🤣

Comment (kobergj) [reply services/postprocessing/pkg/config/config.go:26]: True. The word `upload` is confusing here. Better with `postprocessing`?

Comment (mmattel) [reply services/postprocessing/pkg/config/config.go:26]: Lets discuss this with a call after Christmas. I dont have a brain for this now.

File: services/postprocessing/pkg/config/config.go [hunk 3]
```diff
@@ -23,16 +23,15 @@ type Config struct {
...
+	Endpoint string `yaml:"endpoint" env:"POSTPROCESSING_EVENTS_ENDPOINT" desc:"The address of the event system. The event system is the message queuing service. It is used as message broker for the microservice architecture."`
+	Cluster  string `yaml:"cluster" env:"POSTPROCESSING_EVENTS_CLUSTER" desc:"The clusterID of the event system. The event system is the message queuing service. It is used as message broker for the microservice architecture. Mandatory when using NATS as event system."`
 
-	TLSInsecure          bool   `yaml:"tls_insecure" env:"OCIS_INSECURE;POSTPROCESSING_EVENTS_TLS_INSECURE" desc:"Whether to verify the server TLS certificates."`
-	TLSRootCACertificate string `yaml:"tls_root_ca_certificate" env:"POSTPROCESSING_EVENTS_TLS_ROOT_CA_CERTIFICATE" desc:"The root CA certificate used to validate the server's TLS certificate. If provided POSTPROCESSING_EVENTS_TLS_INSECURE will be seen as false."`
-	EnableTLS            bool   `yaml:"enable_tls" env:"OCIS_EVENTS_ENABLE_TLS;POSTPROCESSING_EVENTS_ENABLE_TLS" desc:"Enable TLS for the connection to the events broker. The events broker is the ocis service which receives and delivers events between the services.."`
```
Comment (kobergj) [inline services/postprocessing/pkg/config/config.go:37]: Please readd this line

Comment (mmattel) [reply services/postprocessing/pkg/config/config.go:37]: fixed

**Review by kobergj — APPROVED** [review]

**Pull request merged**